tests/e2e/support/objects/runtime/application.ts

```diff
@@ -74,4 +75,77 @@ export class Application {
       await this.#page.locator(closeSidebarBtn).click()
     }
   }
+
+  async waitForTokenRenewalViaRefreshToken(): Promise<void> {
+    await Promise.all([
+      this.#page.waitForResponse(
+        (resp) => !resp.url().includes('/oidc-silent-redirect.html'),
+        // timeout after 50 seconds
+        { timeout: (config.timeout - 10) * 1000 }
+      ),
+      this.#page.waitForResponse(
+        (resp) =>
+          resp.url().includes('/token') &&
+          resp.status() === 200 &&
+          resp.request().method() === 'POST' &&
+          resp.request().postDataJSON().grant_type === 'refresh_token' &&
+          resp.request().postDataJSON().hasOwnProperty('refresh_token') &&
+          resp.request().postDataJSON().refresh_token &&
+          resp.request().postDataJSON().hasOwnProperty('scope') &&
+          resp.request().postDataJSON().scope.includes('offline_access'),
+        // timeout after 50 seconds
+        { timeout: (config.timeout - 10) * 1000 }
+      )
+    ])
+  }
```

Comment:
we added this function to check that the token renewal was done using the refresh_token.